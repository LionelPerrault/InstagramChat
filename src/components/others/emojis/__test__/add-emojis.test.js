import React from 'react'
import { create } from 'react-test-renderer'
import AddEmojis from '../add-emojis'
import { shallow } from 'enzyme'

describe('Add-Emojis Component', () => {
  const props = {
    position: {
      top: 10,
      left: 50,
    },
    textArea: '.some-textarea',
    updateTextArea() {},
  }

  it('should match snapshot', () => {
    const tree = create(<AddEmojis {...props} />).toJSON()
    expect(tree).toMatchSnapshot()

    const tree2 = create(
      <AddEmojis
        {...props}
        disabled
        addClassOnClicked
        className=".emoji-modal-shown"
      />
    ).toJSON()
    expect(tree2).toMatchSnapshot()
  })

  it('should toggle Emojis Component on click', () => {
    const wrapper = shallow(<AddEmojis {...props} />)

    expect(wrapper.state().showEmojis).toBeFalse()
    expect(wrapper.find('Emojis').exists()).toBeFalse()

    wrapper.find('span').simulate('click')

    expect(wrapper.find('Emojis').exists()).toBeTrue()
  })
})
