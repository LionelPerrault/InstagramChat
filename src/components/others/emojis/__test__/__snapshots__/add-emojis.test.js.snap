// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add-Emojis Component should match snapshot 1`] = `
Array [
  <span
    className="emoji_span"
    data-tip="Add emojis"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      sentiment_very_satisfied
    </i>
  </span>,
  <div
    className="__react_component_tooltip place-top type-dark "
    data-id="tooltip"
  >
    
  </div>,
]
`;

exports[`Add-Emojis Component should match snapshot 2`] = `
Array [
  <span
    className="emoji_span emoji_disabled"
    data-tip="Add emojis"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      sentiment_very_satisfied
    </i>
  </span>,
  <div
    className="__react_component_tooltip place-top type-dark "
    data-id="tooltip"
  >
    
  </div>,
]
`;
